src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -1027,9 +1046,15 @@ def compute_several_extensions(self, extensions, save=True, verbose=False, **job
         >>> sorting_analyzer.compute_several_extensions({"waveforms": {"ms_before": 1.2}, "templates" : {"operators": ["average", "std"]}})
 
         """
+        extensions_to_recompute = {}
+        if not force_recompute:
+            for extension_name, extension_params in extensions.items():
+                if self.does_extension_need_recompute(extension_name, **extension_params):
+                    extensions_to_recompute[extension_name] = extension_params
```

Comment:
Hello, there's a problem here.
If I do something like:
```
sorting_analyzer.compute(["random_spikes", "waveforms", "templates", "spike_locations"])
sorting_analyzer.compute(["random_spikes", "waveforms", "templates", "spike_locations"], 
    extension_params={
        'waveforms': {'ms_before': 1.5},
        'spike_locations' : {'ms_after': 0.4}
    }
)
```
Then it only includes `waveforms` and `spike_locations` in the recomputation. But we should also recompute templates. So you need to include this too. Something like:
```
if self.does_extension_need_recompute(extension_name, **extension_params):
    extensions_to_recompute[extension_name] = extension_params
        for ext_dependency in set(_get_children_dependencies(extension_name)) & set(extensions.keys()):
            extensions_to_recompute[ext_dependency] = extensions[ext_dependency]
```